index.html

```diff
@@ -204,55 +229,2313 @@ <h3>
+                |device|.{{HIDDevice/productId}}, then return `false`.
+                </li>
+              </ol>
+            </li>
+            <li>Return `true`.
+            </li>
+          </ol>
+          <p>
+            A {{HIDDeviceFilter}} |filter:HIDDeviceFilter| <dfn>matches any
+            collection</dfn> of |device:HIDDevice| if these steps return
+            `true`:
+          </p>
+          <ol>
+            <li>[=list/For each=] |collection:HIDCollectionInfo| of
+            |device|.{{HIDDevice/collections}}:
+              <ol>
+                <li>If |filter| [=matches the collection=] |collection| then
+                return `true`.
+                </li>
+              </ol>
+            </li>
+            <li>Return `false`.
+            </li>
+          </ol>
+          <p>
+            A {{HIDDeviceFilter}} |filter:HIDDeviceFilter| <dfn>matches the
+            collection</dfn> |collection:HIDCollectionInfo| if these steps
+            return `true`:
+          </p>
+          <ol>
+            <li>If "{{HIDDeviceFilter/usagePage}}" is in |filter|:
+              <ol>
+                <li>If |filter|["{{HIDDeviceFilter/usagePage}}"] is not equal
+                to |collection|.{{HIDCollectionInfo/usagePage}}, then return
+                `false`.
+                </li>
+                <li>If "{{HIDCollectionInfo/usage}}" is in |filter| and
+                |filter|["{{HIDDeviceFilter/usage}}"] is not equal to
+                |collection|.{{HIDCollectionInfo/usage}}, then return `false`.
+                </li>
+              </ol>
+            </li>
+            <li>Return `true`.
+            </li>
+          </ol>
+        </section>
+      </section>
+    </section>
+    <section data-dfn-for="HIDDevice">
+      <h2>
+        `HIDDevice` interface
+      </h2>
+      <pre class="idl">
+        [Exposed=Window, SecureContext]
+        interface HIDDevice : EventTarget {
+            attribute EventHandler oninputreport;
+            readonly attribute boolean opened;
+            readonly attribute unsigned short vendorId;
+            readonly attribute unsigned short productId;
+            readonly attribute DOMString productName;
+            readonly attribute FrozenArray&lt;HIDCollectionInfo&gt; collections;
+            Promise&lt;undefined&gt; open();
+            Promise&lt;undefined&gt; close();
+            Promise&lt;undefined&gt; sendReport([EnforceRange] octet reportId, BufferSource data);
+            Promise&lt;undefined&gt; sendFeatureReport(
+                [EnforceRange] octet reportId,
+                BufferSource data);
+            Promise&lt;DataView&gt; receiveFeatureReport([EnforceRange] octet reportId);
         };
       </pre>
+      <p>
+        Methods on this interface typically complete asynchronously, queuing
+        work on the <dfn>HID device task source</dfn>.
+      </p>
+      <p>
+        The [=get the parent=] algorithm for {{HIDDevice}} returns the same
```

Comment:
The primary use case for the "get the parent" algorithm is supporting bubbling for events, which we currently don't enable for any of the events fired by the existing algorithms. It might be nice to turn this on. In particular it allowed `disconnect` event handlers in the Web Serial API to be added to either a particular `SerialPort` instance or `navigator.serial`. It's less useful for `input` events, though that would make the API similar to the [`IOHIDManagerRegisterInputReportCallback`](https://developer.apple.com/documentation/iokit/1438397-iohidmanagerregisterinputreportc) method on macOS.

Reply:
The disconnect event handler case sounds useful. I've removed get-the-parent, let's leave it out for now and add it back along with bubbling.